File: 597668BA-4EFC-4254-A0A6-DADD53C55DE9/B7ADE3AE-1FE8-4D2B-B6C5-8F6D5BDC7F84/CounterService.swift
struct CounterService {
    private let config: CounterConfig

    init(config: CounterConfig) {
        precondition(config.from <= config.to, "\"from\" (\(config.from)) must be <= \"to\" (\(config.to))")
        self.config = config
    }

    func count() -> [CounterOutput] {
        (config.from...config.to).map { i in
            CounterOutput(value: i, label: "Count: \(i)")
        }
    }
}

File: 80664960-AD7B-4DD2-90E7-65AC8E0F1847/553F9A43-F3F6-4DAA-95DD-1DFCB021265D/CounterService.kt
class CounterService(private val config: CounterConfig) {
    init {
        require(config.from <= config.to) {
            "\"from\" (${config.from}) must be <= \"to\" (${config.to})"
        }
    }

    fun count(): Sequence<CounterOutput> = sequence {
        for (i in config.from..config.to) {
            yield(CounterOutput(value = i, label = "Count: $i"))
        }
    }
}
